samples/ControlCatalog.Desktop/Program.cs

```diff
@@ -10,6 +10,7 @@ namespace ControlCatalog
 {
     internal class Program
     {
+        [STAThread]
```

Comment:
Does this not work on .NET core? At the moment our templates assume one can use the same template across .NET Core and .NET Framework.

Reply:
This PR makes me think it should work on .NET core too: https://github.com/dotnet/coreclr/pull/15652

Reply:
It does not work in the current version but as you mentioned the next release will take the attribute into account. For now only `Thread.CurrentThread.TrySetApartmentState(ApartmentState.STA);` will help on .net core

Reply:
Couldn't this be set inside the appbuilder, so that doesn't have to be marked in user code?

Reply:
[MSDN](https://msdn.microsoft.com/en-US/library/system.threading.thread.trysetapartmentstate(v=vs.110).aspx) says it needs to be done before the thread is started. might be possible but I guess Its to late.

What might be a possibility: Only register the Windows D+D implementation if the current thread is STA. That way "In-Process" D+D would always work and "Cross-Process" D+D would work on OSX even if the user did not set the ApartmentState.